src/main/mw.c

```diff
@@ -686,6 +690,110 @@ void processRx(void)
 
 }
 
+void getArmingChannel(modeActivationCondition_t *modeActivationConditions, uint8_t *armingChannel) {
+    for (int index = 0; index < MAX_MODE_ACTIVATION_CONDITION_COUNT; index++) {
+        modeActivationCondition_t *modeActivationCondition = &modeActivationConditions[index];
+    	if (modeActivationCondition->modeId) {
```

Comment:
Strange ... `modeId` for arming channel should be zero, so it should not trigger the if.

IMO this is correxct way to do it: https://github.com/cleanflight/cleanflight/blob/master/src/main/io/rc_controls.c#L734